src/Analyser/TypeSpecifier.php

```diff
@@ -1105,6 +1105,8 @@ private function specifyTypesForCountFuncCall(
 			if (
 				$sizeType instanceof IntegerRangeType
 				&& $sizeType->getMin() !== null
+				&& $sizeType->getMin() <= ConstantArrayTypeBuilder::ARRAY_COUNT_LIMIT
```

Comment:
Do we need another test case for the conditions?

Reply:
does not hurt for sure. I adapted things slightly.